File: 991. Broken Calculator/main.go
func brokenCalc(startValue int, target int) int {
    res := 0
    val := target

    if startValue >= target {
        return startValue - target
    }

    for val != startValue {
        if val % 2 == 0 && val > startValue {
            val = val / 2
        } else {
            val = val + 1
        }
        res = res + 1
    }

    return res
}

File: 991. Broken Calculator/main.ts
function brokenCalc(startValue: number, target: number): number {
    let res = 0
    let val = target

    if (startValue >= target) {
        return startValue - target
    }

    while (val !== startValue) {
        // console.log(val, res)
        if (val % 2 === 0 && val > startValue) {
            val /= 2
        } else {
            ++val
        }
        ++res
    }

    return res
};
